File: String/Easy/1221. Split a String in Balanced Strings.go
package main

import (
	"fmt"
	"strings"
)

func main() {

	str := "RLRRLLRLRL"
	result := balancedStringSplit(str)
	fmt.Println(result)
}

func balancedStringSplit(s string) int {
	count, str := 0, strings.Split(s, "")

	temp := str[0]
	for i := 1; i < len(str); i++ {
		temp += str[i]
		if isBalanced(temp) {
			count++
		}
	}

	return count
}

func isBalanced(s string) bool {
	rCount, lCount, str := 0, 0, strings.Split(s, "")
	for _, val := range str {
		if val == "R" {
			rCount++
		} else if val == "L" {
			lCount++
		}
	}

	if rCount == lCount {
		return true
	}

	return false
}

File: Miscellaneous/Python/1221. Split a String in Balanced Strings.py
class Solution:
    def balancedStringSplit(self, s):
        charCount, result = 0, 0

        for char in s:
            if char == "L":
                charCount += 1
            else:
                charCount -= 1
            if charCount == 0:
                result += 1
        
        return result


obj = Solution()
s = "RLRRLLRLRL"
ans = obj.balancedStringSplit(s)
print(ans)